app/policies/user_subscription_policy.rb

```diff
@@ -0,0 +1,15 @@
+class UserSubscriptionPolicy < ApplicationPolicy
+  def create?
+    user.present?
+  end
+
+  def destroy?
+    create? && [record.subscriber_id, record.subscription_id].include?(user.id)
```

Comment:
Same comments as in other policies + `[record.subscriber_id, record.subscription_id].include?(user.id)` has to be split into multiple more "self-explanatory" methods